### PR TITLE
feat(type): worker registration library

### PR DESCRIPTION
**Pull request by timhuynh94**

### CODE REVIEW

File: library/worker_registration_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+			t.Errorf("GetQueueAddress is %v, want %v", test.qR.GetQueueAddress(), test.want.GetQueueAddress())
+		}
+
+		if test.qR.GetPublicKey() != test.want.GetPublicKey() {
+			t.Errorf("GetPublicKey is %v, want %v", test.qR.GetPublicKey(), test.want.GetPublicKey())
+		}
+	}
+}
+
+// testWorker is a test helper function to create a Worker
```
Comment (ecrupper) [inline library/worker_registration_test.go:82]:
```suggestion
// testQueueRegistration is a test helper function to create a QueueRegistration
```

**Review by ecrupper — COMMENTED** [review]
Tiny thing I saw

File: library/worker_registration.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	RegistrationToken *string `json:"registration_token,omitempty"`
+	QueuePublicKey    *string `json:"queue_public_key,omitempty"`
+	QueueAddress      *string `json:"queue_address,omitempty"`
+}
+
+// GetRegistrationToken returns the RegistrationToken field.
+//
+// When the provided Worker type is nil, or the field within
+// the type is nil, it returns the zero value for the field.
+func (q *WorkerRegistration) GetRegistrationToken() string {
```
Comment (plyr4) [inline library/worker_registration.go:20]: small nitpick, should we change `q` to like, `r` for "Registration"?

Comment (plyr4) [reply library/worker_registration.go:20]: (for all of them)

File: library/worker_registration_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	}
+}
+
+// testWorkerRegistration is a test helper function to register a Worker
+// type with all fields set to a fake value.
+func testWorkerRegistration() *WorkerRegistration {
+	w := new(WorkerRegistration)
+	w.SetRegistrationToken("1234356")
+	w.SetPublicKey("http://localhost:8080")
+	w.SetPublicKey("isfnw1234")
```
Comment (wass3r) [inline library/worker_registration_test.go:88]: I'm guessing the first one should be for queue address?

File: library/worker_registration.go [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+// swagger:model WorkerRegistration
+type WorkerRegistration struct {
+	RegistrationToken *string `json:"registration_token,omitempty"`
+	QueuePublicKey    *string `json:"queue_public_key,omitempty"`
+	QueueAddress      *string `json:"queue_address,omitempty"`
+}
+
+// GetRegistrationToken returns the RegistrationToken field.
+//
+// When the provided Worker type is nil, or the field within
```
Comment (wass3r) [inline library/worker_registration.go:18]:
```suggestion
// When the provided WorkerRegistration type is nil, or the field within
```
Probably from copy pasta from the Worker type. Should be changed in a number of places here. I'm only marking this instance.

**Review by wass3r — CHANGES_REQUESTED** [review]

File: library/worker_registration_test.go (hunk 3 above)
Comment (timhuynh94) [reply library/worker_registration_test.go]: yes!

File: library/worker_registration.go [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	QueuePublicKey    *string `json:"queue_public_key,omitempty"`
+	QueueAddress      *string `json:"queue_address,omitempty"`
+}
+
+// GetRegistrationToken returns the RegistrationToken field.
+//
+// When the provided WorkerRegistration type is nil, or the field within
+// the type is nil, it returns an empty string for the field.
+func (w *WorkerRegistration) GetRegistrationToken() string {
+	// return zero value if Worker type or ID field is nil
```
Comment (wass3r) [inline library/worker_registration.go:21]:
```suggestion
	// return zero value if WorkerRegistration type or RegistrationToken field is nil
```

File: library/worker_registration.go [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+	return *w.RegistrationToken
+}
+
+// GetPublicKey returns the QueuePublicKey field.
+//
+// When the provided WorkerRegistration type is nil, or the field within
+// the type is nil, it returns an empty string for the field.
+func (w *WorkerRegistration) GetPublicKey() string {
+	// return zero value if Worker type or ID field is nil
```
Comment (wass3r) [inline library/worker_registration.go:34]:
```suggestion
	// return zero value if WorkerRegistration type or QueuePublicKey field is nil
```

File: library/worker_registration.go [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+	return *w.QueuePublicKey
+}
+
+// GetQueueAddress returns the QueueAddress field.
+//
+// When the provided WorkerRegistration type is nil, or the field within
+// the type is nil, it returns an empty string for the field.
+func (w *WorkerRegistration) GetQueueAddress() string {
+	// return zero value if Worker type or ID field is nil
```
Comment (wass3r) [inline library/worker_registration.go:47]:
```suggestion
	// return zero value if WorkerRegistration type or QueueAddress field is nil
```

File: library/worker_registration.go [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+
+	return *w.QueueAddress
+}
+
+// SetRegistrationToken sets the RegistrationToken field.
+//
+// When the provided WorkerRegistration type is nil, it
+// will set nothing and immediately return.
+func (w *WorkerRegistration) SetRegistrationToken(v string) {
+	// return if Worker type is nil
```
Comment (wass3r) [inline library/worker_registration.go:60]:
```suggestion
	// return if WorkerRegistration type is nil
```

File: library/worker_registration.go [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+	w.RegistrationToken = &v
+}
+
+// SetPublicKey sets the QueuePublicKey field.
+//
+// When the provided WorkerRegistration type is nil, it
+// will set nothing and immediately return.
+func (w *WorkerRegistration) SetPublicKey(v string) {
+	// return if Worker type is nil
```
Comment (wass3r) [inline library/worker_registration.go:73]:
```suggestion
	// return if WorkerRegistration type is nil
```

File: library/worker_registration.go [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+	w.QueuePublicKey = &v
+}
+
+// SetQueueAddress sets the QueueAddress field.
+//
+// When the provided WorkerRegistration type is nil, it
+// will set nothing and immediately return.
+func (w *WorkerRegistration) SetQueueAddress(v string) {
+	// return if Worker type is nil
```
Comment (wass3r) [inline library/worker_registration.go:86]:
```suggestion
	// return if WorkerRegistration type is nil
```

**Review by wass3r — CHANGES_REQUESTED** [review]
few more place to update

**Review by wass3r — APPROVED** [review]

**Review by plyr4 — APPROVED** [review]

**Pull request merged**